### PR TITLE
Remove IRModule Dependency from Target

### PR DESCRIPTION
**Pull request by yongwww**
There is a dependency of IRModule in the current Target implementation: `target.h -> module.h`. This pull request aims to decouple the dependency."

### CODE REVIEW

File: include/tvm/relay/transform.h [hunk 1]
```diff
@@ -47,6 +47,18 @@ using PassInfoNode = tvm::transform::PassInfoNode;
 using PassContext = tvm::transform::PassContext;
 using PassContextNode = tvm::transform::PassContextNode;
 using Sequential = tvm::transform::Sequential;
+using FTVMRelayToTIR = tvm::transform::Pass;
```
Comment (ekalda) [inline include/tvm/relay/transform.h:50]: Looks like this one has lost its documentation during the refactor...

Comment (yongwww) [reply include/tvm/relay/transform.h]: added, thanks!

File: include/tvm/relay/transform.h [hunk 2]
```diff
@@ -47,6 +47,18 @@ using PassInfoNode = tvm::transform::PassInfoNode;
...
+ * \brief TIRToRuntime conversion specific to a TargetKind
+ *
+ * This function is responsible for scanning an IRModule for appropriate Target-specific functions
+ and generating a Runtime module representing the compiled output
+ *
+ * \param ir_module Unified IRModule
+ * \param target Target to filter on or retrieve arguments from
+ * \return Runtime Module containing compiled functions
+ */
+using FTVMTIRToRuntime = tvm::runtime::TypedPackedFunc<runtime::Module(IRModule, Target)>;
```
[the remark this answers is not in the document]
Comment (yongwww) [reply include/tvm/relay/transform.h]: Thanks for the inputs! I changed the code to directly use "`using FTVMTIRToRuntime = tvm::runtime::TypedPackedFunc<runtime::Module(IRModule, Target)>`;" in target.cc & codegen.cc.

**Review by kparzysz-quic — APPROVED** [review]

**Comment by yongwww** [comment]
cc: @tqchen @junrushao

**Pull request merged**